src/diagnostics/MOM_spatial_means.F90

```diff
@@ -202,10 +213,7 @@ function global_area_integral(var, G, scale, area, tmp_scale, unscale)
     enddo ; enddo
   endif
 
-  global_area_integral = reproducing_sum(tmpForSumming)
-
-  if ((temp_scale /= 0.0) .and. (temp_scale /= 1.0)) &
-    global_area_integral = global_area_integral / temp_scale
+  global_area_integral = reproducing_sum(tmpForSumming, unscale=temp_scale)
```

Comment:
Not that it matters in any meaningful way, but is there any reason that the areal unscaling (`G%US%L_to_m**2`) is not performed *within* `reproducing_sum` when `tmp_scale` is not present, unlike the other functions? I see that area can be from an optional argument here, but not sure that's why.

Reply:
Yes, the integrated area is also rescaled from [L2 ~> m2] to [m2] in the returned variable when `tmp_scale` is not present.  This is deliberate and functionally important.